Source/Core/Core/AchievementManager.cpp

```diff
@@ -674,111 +247,58 @@ void AchievementManager::DoFrame()
-    m_update_callback();
+    rc_client_get_rich_presence_message(m_client, m_rich_presence.data(), RP_SIZE);
+    m_update_callback(UpdatedItems{.rich_presence = true});
   }
 }
 
-u32 AchievementManager::MemoryPeeker(u32 address, u32 num_bytes, void* ud)
-{
-  if (!m_system)
-    return 0u;
-  Core::CPUThreadGuard threadguard(*m_system);
-  switch (num_bytes)
-  {
-  case 1:
-    return m_system->GetMMU()
-        .HostTryReadU8(threadguard, address, PowerPC::RequestedAddressSpace::Physical)
-        .value_or(PowerPC::ReadResult<u8>(false, 0u))
-        .value;
-  case 2:
-    return Common::swap16(
-        m_system->GetMMU()
-            .HostTryReadU16(threadguard, address, PowerPC::RequestedAddressSpace::Physical)
-            .value_or(PowerPC::ReadResult<u16>(false, 0u))
-            .value);
-  case 4:
-    return Common::swap32(
-        m_system->GetMMU()
-            .HostTryReadU32(threadguard, address, PowerPC::RequestedAddressSpace::Physical)
-            .value_or(PowerPC::ReadResult<u32>(false, 0u))
-            .value);
-  default:
-    ASSERT(false);
-    return 0u;
-  }
-}
-
-void AchievementManager::AchievementEventHandler(const rc_runtime_event_t* runtime_event)
+std::recursive_mutex& AchievementManager::GetLock()
 {
-  switch (runtime_event->type)
-  {
-  case RC_RUNTIME_EVENT_ACHIEVEMENT_TRIGGERED:
-    HandleAchievementTriggeredEvent(runtime_event);
-    break;
-  case RC_RUNTIME_EVENT_ACHIEVEMENT_PROGRESS_UPDATED:
-    HandleAchievementProgressUpdatedEvent(runtime_event);
-    break;
-  case RC_RUNTIME_EVENT_ACHIEVEMENT_PRIMED:
-    HandleAchievementPrimedEvent(runtime_event);
-    break;
-  case RC_RUNTIME_EVENT_ACHIEVEMENT_UNPRIMED:
-    HandleAchievementUnprimedEvent(runtime_event);
-    break;
-  case RC_RUNTIME_EVENT_LBOARD_STARTED:
-    HandleLeaderboardStartedEvent(runtime_event);
-    break;
-  case RC_RUNTIME_EVENT_LBOARD_CANCELED:
-    HandleLeaderboardCanceledEvent(runtime_event);
-    break;
-  case RC_RUNTIME_EVENT_LBOARD_TRIGGERED:
-    HandleLeaderboardTriggeredEvent(runtime_event);
-    break;
-  }
+  return m_lock;
 }
 
-std::recursive_mutex& AchievementManager::GetLock()
+void AchievementManager::SetHardcoreMode()
 {
-  return m_lock;
+  rc_client_set_hardcore_enabled(m_client, Config::Get(Config::RA_HARDCORE_ENABLED));
 }
 
 bool AchievementManager::IsHardcoreModeActive() const
 {
   std::lock_guard lg{m_lock};
-  if (!Config::Get(Config::RA_HARDCORE_ENABLED))
+  if (!rc_client_get_hardcore_enabled(m_client))
     return false;
-  if (!Core::IsRunning())
+  if (!rc_client_get_game_info(m_client))
     return true;
-  if (!IsGameLoaded())
-    return false;
-  return (m_runtime.trigger_count + m_runtime.lboard_count > 0);
+  return rc_client_is_processing_required(m_client);
+}
+
+void AchievementManager::SetSpectatorMode()
+{
+  rc_client_set_spectator_mode_enabled(m_client, Config::Get(Config::RA_SPECTATOR_ENABLED));
 }
 
 std::string AchievementManager::GetPlayerDisplayName() const
```

Comment:
This as well as `GetGameDisplayName` can return `std::string_view` to avoid string conversions using `QtUtils::FromStdString` as well.